src/bos/reporter/client.py

```diff
@@ -71,12 +71,29 @@ def get_auth_token(path='/opt/cray/auth-utils/bin/get-auth-token'):
         time.sleep(2)
 
 
-def requests_retry_session(retries=10, connect=10, backoff_factor=0.5,
-                           status_forcelist=(500, 502, 503, 504),
-                           session=None):
+class TimeoutHTTPAdapter(HTTPAdapter):
     """
-    Returns a session with retries built into it.
+    An HTTP Adapter that allows a session level timeout for both read and connect attributes. This prevents interruption
```

Comment:
This is the same code as in https://github.com/Cray-HPE/bos/pull/339/files#:~:text=src/bos/operators/utils/__init__.py. 
Don't repeat yourself. (DRY)

Reply:
This was intentional. In the CSM 1.6 branch I did remove the duplication. But in CSM 1.4, I wanted to minimize the changes made. And changing BOS reporter to just import the function from the operators code breaks the RPM, unless other changes are made to the spec file to account for it (see [CASMTRIAGE-7019](https://jira-pro.it.hpe.com:8443/browse/CASMTRIAGE-7019) for details). I wanted to avoid making that extensive of a change in a support branch.

(I'll also note that BOS reporter already has other instances where it duplicates code that exists elsewhere in the repo -- I've corrected this in `develop`, but again, didn't feel it made sense to correct it for this support branch).